### DISCUSSION TITLE
deterministic GenGaloisKeysNew?

### DISCUSSION THREAD

**Discussion by fedejinich** [tuneinsight/lattigo#397] [Q&A] [answered]
Hi, I'm using [this](https://github.com/tuneinsight/lattigo/tree/dev_bgvfv) lattigo version, and I wanted to check about this situation

```go
a := keygen.GenGaloisKeyNew(e, secretKey)
b := keygen.GenGaloisKeyNew(e, secretKey)

if !a.Equal(b) {
  panic("this is unexpected")
}
```

It surprised me that `a` is not equal to `b` (given the same `galEl` and `secretKey`). Is that expected? Is there any way to generate galois keys in a deterministic way (Ideally without marshalling into binary)

**Comment by ChristianMct** [comment, marked as answer]
This is an expected behavior. All (centralized) public-key generation functions are sampling "fresh" RLWE samples.  I.e., they sample fresh public polynomial vector `a` and fresh secret error vector `e` at each call, so they output fresh keys `(sa + X + e, a)` (for `X` a variable that depends on the type of public key). There is *currently* no easy way to obtain a deterministic generation. The only way would be to re-implement the key-generation algo from the `ring` package, with seeded samplers.

That said, we are aware that this can be a problem, and we are planning to add the possibility of seeding the PRNGs behind our API.

**Reply by fedejinich** [reply]
Okay, thanks for the answer; very clear!
